src/main/java/com/google/sps/data/DeliverySystem.java

```diff
@@ -26,16 +26,74 @@
+    Journey journey = findOrdersForDeliveryRequest(deliverySlot);
+    return journey.findOptimalOrderForWaypoints();
+  }
+
+  /**
+   * Assigns a set of orders for the delivery request such that they fit the [startTime, endTime]
+   * timeslot.
+   */
+  public Journey findOrdersForDeliveryRequest(DeliverySlot deliverySlot)  throws ApiException, BadRequestException, DataNotFoundException, IOException, InterruptedException {
+    Point startPoint = deliverySlot.getStartPoint();
+    Journey journey = new Journey(startPoint, pathFinder);
+    OrderHandler orderHandler = new OrderHandler(pathFinder);
+     // Get unassigned orders from datastore which are in the area of the starting point.
+    List<Entity> orders = orderHandler.getAvailableOrders(startPoint.getArea());
+    for (Entity order : orders) {
+      if (journey.getNumberOfWaypoints() >= MAX_WAYPOINTS) {
+        break;
+      }
+
+      LibraryPoint library = new LibraryPoint((double) order.getProperty(OrderHandler.OrderProperty.LIBRARY_LAT.label), 
+        (double) order.getProperty(OrderHandler.OrderProperty.LIBRARY_LNG.label),
+        (int) order.getProperty(OrderHandler.OrderProperty.LIBRARY_ID.label));
+      Point recipient = new Point((double) order.getProperty(OrderHandler.OrderProperty.RECIPIENT_LAT.label), 
+        (double) order.getProperty(OrderHandler.OrderProperty.RECIPIENT_LNG.label));
+      
+      // Check if the library is already a waypoint in the journey, and add it in case it's not.
+      boolean libraryIsWaypoint = !journey.addPointToWaypoints(library);
+      // Check if the recipient is already a waypoint in the journey, and add it in case it's not.
+      boolean recipientIsWaypoint = !journey.addPointToWaypoints(recipient);
+
+      if (journey.getNumberOfWaypoints() > MAX_WAYPOINTS || !journey.findJourneyForTimeslot(deliverySlot)) {
+        // If the new journey doesn't satisfy user's prefrences, remove the order.
+        if (!libraryIsWaypoint) {
+          // Remove the library from waypoints if it was added only for this order.
+          journey.removeWaypoint(library);
+        }
+        if (!recipientIsWaypoint) {
+          // Remove the recipient from the waypoints if he has no other order assigned to this 
+          // delivery slot.
+          journey.removeWaypoint(recipient);
+        }
+      } else {
+        // The library must be visited before the recipient.
+        journey.addRestriction(journey.getWaypointIndex(library), journey.getWaypointIndex(recipient));
+        // Add the books from current order to the library their are assigned to.
+        library.addBooks((List<String>)order.getProperty("books"));
+        // The order is assigned to the delivery slot and its status changes in datastore.
+        order.setProperty("status", "ASSIGNED");
```

Comment:
You really don't want to do perform database mutations in a method called `find`. That will confuse everyone. It's also a really bad idea to tie your algorithm logic with your database mutations.

Do this instead:
- Run your algorithm, pick your waypoints, do all your sanity checks and very that everything is fine, whatever it is you need to do. <= ideally this class would only do this.
- Then commit your mutations to the database: create the journey for the courier in the database, mark the orders as assigned to the journey, etc.

Ideally the commit part should and in a transaction (https://cloud.google.com/datastore/docs/concepts/transactions) to prevent an error from actually committing invalid state to the database (e.g. the order is marked as assigned, but noone is actually delivery it). But a TODO is totally fine for now.

Having your algorithm perform database mutations "as it goes" has several disadvantages:
- Hard to test. Your tests not only need to test the final output, but also database mutations (you don't seem to be doing that in your tests, btw)
- Hard to move to use a transaction: mutations happen all over the place in many classes. Even if you do manage to make them work, your transactions would be horribly inefficient. You generally want transactions to be quick, without having to have a transaction live for the life of a potentially slow running algorithm.
- Hard to refactor: I want to change the database, now I need to have a deep understanding of the routing algorithm... I want to change the algorithm, I need to understand datastore schema... why?

Reply:
I understand, I changed the class so that the status of orders is updated in OrderHandler.java. In the next PR I will use transactions for updates and remove the database logic from DeliverySystem.

Reply:
Add a TODO somewhere.

Reply:
Done.